librz/debug/p/native/linux/linux_debug.c

```diff
@@ -922,6 +923,17 @@ RzList /*<RzDebugPid *>*/ *linux_thread_list(RzDebug *dbg, int pid, RzList /*<Rz
 
 			rz_debug_reg_sync(dbg, RZ_REG_TYPE_GPR, false);
 			pc = rz_debug_reg_get(dbg, "PC");
+			RzRegItem *ri = rz_reg_get(dbg->reg, "fs", RZ_REG_TYPE_ANY);
+			RZ_DEBUG_REG_T regs;
+			// Fetch gs_base from a ptrace call
+			if (ri == NULL && !strcmp(dbg->arch, "x86")) {
+				if (rz_debug_ptrace(dbg, PTRACE_GETREGS, dbg->tid, NULL, &regs) != -1) {
+					tls = regs.gs_base;
+				}
```

Comment:
if the first `if` is true but the second `if` is false, `tls` is `0`, but is that ok ?

Reply:
Added a comment to handle this from caller, this would be set to `0`